app/src/main/java/ayds/newyork/songinfo/home/view/DateFormatter.kt

```diff
@@ -0,0 +1,71 @@
+package ayds.newyork.songinfo.home.view
+
+import ayds.newyork.songinfo.home.model.repository.local.spotify.DatePrecision
+import java.text.SimpleDateFormat
+import java.util.*
+
+private const val DEFAULT_RESULT = "N/A"
+private const val NEW_MONTH_FORMAT = "MMMM, yyyy"
+private const val NEW_DAY_FORMAT = "dd/MM/yyyy"
+
+class DateFormatter {
+
+    fun getReleaseDate(date: String, precisionDate: DatePrecision) =
+        when (precisionDate) {
+            DatePrecision.DAY -> getDayFormat(date)
+            DatePrecision.MONTH -> getMonthFormat(date)
+            DatePrecision.YEAR -> getYearFormat(date)
+            else -> ""
+        }
+
+
+        private fun getYearFormat(year: String): String {
+            var format = DEFAULT_RESULT
+            if (year.isNotEmpty()) {
+                format =
+                    year + if (isLeapYear(year.toInt())) " (leap year)" else " (not a leap year)"
+            }
+            return format
+        }
+
+
+        private fun isLeapYear(year: Int): Boolean {
+            return when {
+                year % 4 != 0 -> {
+                    false
+                }
+                year % 400 == 0 -> {
+                    true
+                }
+                else -> year % 100 != 0
+            }
+        }
```

Comment:
Se puede simplificar
```suggestion
        private fun isLeapYear(year: Int) = ((year % 4 == 0) && (year % 100 != 0 || year % 400 == 0))
```